src/cloudfront/index.ts

```diff
@@ -141,13 +140,106 @@ export const createCloudFrontDistribution = async (
   await cloudfront
     .waitFor('distributionDeployed', { Id: Distribution.Id })
     .promise()
+
   return Distribution
 }
 
+const createAndPublishNoDefaultRootObjectRedirectionFunction = async () => {
+  const cloudFrontRedirectionFunctionName =
+    NO_DEFAULT_ROOT_OBJECT_REDIRECTION_FUNCTION_NAME +
+    '_' +
+    NO_DEFAULT_ROOT_OBJECT_REDIRECTION_COLOR
+
+    let functionARN = await isCloudFrontFunctionExisting(
+      cloudFrontRedirectionFunctionName
+    )
+
+  if (!functionARN) {
+    const { createdFunctionETag, createdFunctionARN } =
+      await createNoDefaultRootObjectFunction(cloudFrontRedirectionFunctionName)
+
+    if (createdFunctionETag) {
+      await publishCloudFrontFunction(
+        cloudFrontRedirectionFunctionName,
+        createdFunctionETag
+      )
+    }
+
+    if (!createdFunctionARN) {
```

Comment:
Is it possible to have the ARN but not the Etag ? 
From the code this situation would endup having a function created but not published and no error throw

Reply:
I think we'll always have the ETag here 👍🏻